third_party/move/move-model/src/spec_translator.rs

```diff
@@ -704,14 +704,13 @@ impl<'a, 'b, T: ExpGenerator<'a>> ExpRewriterFunctions for SpecTranslator<'a, 'b
                 )
                 .into_exp(),
             ),
-            Exists(None) if self.in_old => Some(
-                Call(
-                    id,
-                    Exists(Some(self.save_memory(self.builder.get_memory_of_node(id)))),
-                    args.to_owned(),
-                )
-                .into_exp(),
-            ),
+            Exists(None) if self.in_old => {
+                if let Some(mem) = self.builder.get_memory_of_node_opt(id) {
+                    Some(Call(id, Exists(Some(self.save_memory(mem))), args.to_owned()).into_exp())
+                } else {
+                    Some(Call(id, Exists(None), args.to_owned()).into_exp())
```

Comment:
Unfortunately, I think this isn't correct. If an `exists` is applied within `old(...exists...)`, we need to create this label to use a different memory. If we do not then the exists will be interpreted within the current memory, which is semantically wrong.

There is no obvious workaround. You could report an error like `exists with type parameter not supported within `old` expression' or similar. Do you need it within old?

Reply:
The issue here is that expressions in the `aborts_if` condition are in the old context and currently the main use for having `exists<T>` is in the aborts_if condition. Do you have any suggestions?

Reply:
It will be difficult. If you check the `Bytecode::SaveMem(label, mem)` instruction, that is what you eventually need to emit here as well. Currently `mem: QualifiedInstId<StructId>`. This would need to be generalized so it can be either a struct id or a type parameter. The generalization would need to be propagated to multiple places. I wonder whether all this is worth it.

Do you see an alternative way to implement `exists_at` just as native function?

Reply:
In my opinion, to implement `exists_at` natively that follows the semantics of memory, we still need to maintain a pair (label, mem) where mem is a type parameter. In that sense, it makes more sense to just extend `Bytecode::SaveMem(label, mem)` in general. I am wondering if `exists_at<T>` is used in for generic types, we can say that `old(...exists<T>)` is always equal to `exists<T>` because global memory cannot be changed if it is a generic type?